src/promptflow/promptflow/_sdk/_record_storage.py

```diff
@@ -0,0 +1,178 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import collections
+import hashlib
+import json
+import os
+from pathlib import Path
+from typing import Any, Dict, OrderedDict
+
+from promptflow._sdk._configuration import Configuration
+from promptflow._sdk._errors import RecordFileMissingException, RecordItemMissingException
+from promptflow._utils.dataclass_serializer import serialize
+from promptflow._utils.tool_utils import get_inputs_for_prompt_template
+
+
+class RecordStorage(object):
+    standard_record_name = ".cache/storage_record.json"
+    _instance = None
+
+    def __init__(self, input_path: str = None):
+        """
+        Status: binding with files inside .promptflow/
+        """
+        #
+        self.input_path: Path = None
+        self._input_path_str: str = None
```

Comment:
consider only keep one form

Reply:
Use getter and setting to sync these values.